release-note-script/src/main/java/ReleaseNoteCreation.java

```diff
@@ -0,0 +1,331 @@
+  }
+
+  private String getFormattedReleaseNotes(List<ReleaseNoteText> releaseNotes) {
+    StringBuilder builder = new StringBuilder();
+    releaseNotes.forEach(
+        rnText -> {
+          builder.append(String.format("- %s (", rnText.text));
+          rnText.prNumbers.forEach(prNum -> builder.append(String.format("#%s ", prNum)));
+          builder.deleteCharAt(builder.length() - 1); // delete the last space character
+          builder.append(")\n");
+        });
+    return builder.toString();
+  }
+
+  private void checkReleaseNoteCategory(ReleaseNoteText rnText) {
+    if (rnText.category == null) {
+      rnText.category = Category.MISCELLANEOUS;
+    }
+  }
+
+  private boolean isContainedInSameAsItems(ReleaseNoteText rnText) {
+    return sameAsItems.values().stream().anyMatch(items -> items.contains(rnText));
+  }
+
+  public enum Category {
+    ENHANCEMENT("Enhancements"),
+    IMPROVEMENT("Improvements"),
+    BUGFIX("Bug fixes"),
+    DOCUMENTATION("Documentation"),
+    MISCELLANEOUS("Miscellaneous");
+
+    private final String displayName;
+
+    Category(String displayName) {
+      this.displayName = displayName;
+    }
+
+    public String getDisplayName() {
+      return this.displayName;
+    }
+
+    public static Category getByName(String name) {
+      return Arrays.stream(Category.values())
+          .filter(v -> v.name().equalsIgnoreCase(name))
+          .findFirst()
+          .orElseThrow(() -> new IllegalArgumentException("Invalid name: " + name));
+    }
+  }
+
+  static class ReleaseNoteText {
+    public Category category;
+    public String text;
+    public List<String> prNumbers = new ArrayList<>();
+  }
+
+  public static class GitHubContext {
+
+    private static final String MERGED_STATE = "merged";
+
+    private final String owner;
+    private final String projectTitlePrefix;
+    private final String version;
+    private final String repository;
+
+    public GitHubContext(String owner, String projectTitString, String version, String repository) {
+      this.owner = owner;
+      this.projectTitlePrefix = projectTitString;
+      this.version = version;
+      this.repository = repository;
+    }
+
+    public String getProjectId() throws Exception {
+      BufferedReader br =
+          runSubProcessAndGetOutputAsReader(
+              format(
+                  "gh project list --owner %s | awk '/%s/ {print}' | awk '/%s/ {print $1}'",
+                  this.owner, this.projectTitlePrefix, this.version));
+
+      String line = br.readLine(); // Assuming only one line exists.
+      if (line == null) throw new RuntimeException("Couldn't get the projectId");
+      return line;
+    }
+
+    public List<String> getPullRequestNumbers(String projectId) throws Exception {
+      BufferedReader br =
+          runSubProcessAndGetOutputAsReader(
+              format(
+                  "gh project item-list %s --owner %s --limit 200 | awk -F'\\t' '/%s\\t/ {print"
```

Comment:
I'm unsure how this script is used, but is it reasonable to assume that the `gh` command is available? I guess it would be better to utilize the GitHub client library in Java instead of launching `gh` as a subprocess.

Reply:
It is assumed that this script is used on GitHub Actions workflow. I think it can be assumed the runtime environment installed `gh` command (I tested it already).

But I understand your point. So I'd like to reorganize this script as the normal java project so that it use useful libraries.

Reply:
No worries. Thank you for your explanation. Given that this script is mainly intended to be used on the GitHub Actions, the assumption of the `gh` command being available is reasonable.